test/getFormat.js

```diff
@@ -1,46 +1,66 @@
+    {"id":"0k2Zzkw_-0I","code":"160","filetype":"mp4","resolution":"192x144","itag":160,"mediaformat":"DASH video","bitrate":"120k","fps":15,"filesize":"3.90MiB"},
+    {"id":"0k2Zzkw_-0I","code":"242","filetype":"webm","resolution":"320x240","itag":242,"mediaformat":"DASH video","bitrate":"207k","fps":30,"filesize":"5.96MiB"},
+    {"id":"0k2Zzkw_-0I","code":"133","filetype":"mp4","resolution":"320x240","itag":133,"mediaformat":"DASH video","bitrate":"248k","fps":30,"filesize":"8.53MiB"},
+    {"id":"0k2Zzkw_-0I","code":"243","filetype":"webm","resolution":"480x360","itag":243,"mediaformat":"DASH video","bitrate":"358k","fps":30,"filesize":"10.21MiB"},
+    {"id":"0k2Zzkw_-0I","code":"134","filetype":"mp4","resolution":"480x360","itag":134,"mediaformat":"DASH video","bitrate":"605k","fps":30,"filesize":"13.12MiB"},
+    {"id":"0k2Zzkw_-0I","code":"244","filetype":"webm","resolution":"640x480","itag":244,"mediaformat":"DASH video","bitrate":"792k","fps":30,"filesize":"20.51MiB"},
+    {"id":"0k2Zzkw_-0I","code":"135","filetype":"mp4","resolution":"640x480","itag":135,"mediaformat":"DASH video","bitrate":"1105k","fps":30,"filesize":"26.55MiB"},
+    {"id":"0k2Zzkw_-0I","code":"17","filetype":"3gp","resolution":"176x144","itag":17},
+    {"id":"0k2Zzkw_-0I","code":"36","filetype":"3gp","resolution":"320x240","itag":36},
+    {"id":"0k2Zzkw_-0I","code":"5","filetype":"flv","resolution":"400x240","itag":5},
+    {"id":"0k2Zzkw_-0I","code":"43","filetype":"webm","resolution":"640x360","itag":43},
+    {"id":"0k2Zzkw_-0I","code":"18","filetype":"mp4","resolution":"640x360","itag":18}
   ],
 
+  crunchyroll: [
+    {"id":"662713","code":"360p","filetype":"flv","resolution":"unknown"},
+    {"id":"662713","code":"480p","filetype":"flv","resolution":"unknown"},
+    {"id":"662713","code":"720p","filetype":"flv","resolution":"unknown"},
+    {"id":"662713","code":"720p","filetype":"flv","resolution":"unknown"},
+    {"id":"662713","code":"1080p","filetype":"flv","resolution":"unknown"},
+    {"id":"662713","code":"1080p","filetype":"flv","resolution":"unknown"}
+   ],
+
   mixcloud: [
     { id: 'TheBloodyBeetroots-sbcr-dj-mix-october-2014', itag: 0, filetype: 'mp3', resolution: 'unknown' }
   ]
 };
 
 vows.describe('getFormats').addBatch({
-  'from a video': {
+  'from a youtube video': {
+    'topic': function() {
+      ytdl.getFormats(youtube, this.callback);
+    },
+
+    'formats returned': function(err, formats) {
+      assert.isNull(err);
+      assert.isArray(formats);
+      assert.deepEqual(formats, expected.youtube);
+    }
+  },
+
+  'from a crunchyroll video': {
     'topic': function() {
-      ytdl.getFormats(video, this.callback);
+      ytdl.getFormats(crunchyroll, this.callback);
     },
 
     'formats returned': function(err, formats) {
       assert.isNull(err);
       assert.isArray(formats);
-      assert.deepEqual(formats, expected.video);
+      assert.deepEqual(formats, expected.crunchyroll);
```

Comment:
I took a quick look at the doc. It would be helpful in that wit prints the diff - but it doesn't look like you can can get the diff as an object and use it for your own purpose. In other words the test itself would fail just the same.